src/pudl/transform/ferc1.py

```diff
@@ -724,10 +765,16 @@ def wide_to_tidy_xbrl(
         later.
         """
         if not params:
-            params = self.params.wide_to_tidy_xbrl
+            # params.wide_to_tidy is not scriptable so we can't extract the "xbrl" or
+            # "dbf" portions dynamically w/o converting to dict
+            params = WideToTidyDataSource(
+                **self.params.wide_to_tidy.dict()[source_ferc1.value]
+            )
```

Comment:
Why hasn't this been necessary in the other cases where we've used the data source to select the appropriate parameters while re-using a single parameter class & function?

Reply:
see `process_xbrl` for the way you did it for `rename_columns`. bc rename columns uses the default `classes` method, we didn't really have the ability to use the convention that we have developed in these transforms which vary by `source_ferc1`. I could change this method to take only a `params` but it would always need to be instantiated version of `WideToTidyFerc1Source` bc i couldn't grab it from `self.params` without knowing which `source_ferc1`